movement/io/load_bboxes.py

```diff
@@ -0,0 +1,644 @@
+"""Load bounding boxes' tracking data into ``movement``."""
+
+import ast
+import logging
+import re
+from collections.abc import Callable
+from pathlib import Path
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from movement import MovementDataset
+from movement.utils.logging import log_error
+from movement.validators.datasets import ValidBboxesDataset
+from movement.validators.files import ValidFile, ValidVIATracksCSV
+
+logger = logging.getLogger(__name__)
+
+
+def from_numpy(
+    position_array: np.ndarray,
+    shape_array: np.ndarray,
+    confidence_array: np.ndarray | None = None,
+    individual_names: list[str] | None = None,
+    frame_array: np.ndarray | None = None,
+    fps: float | None = None,
+    source_software: str | None = None,
+) -> xr.Dataset:
+    """Create a ``movement`` bounding boxes dataset from NumPy arrays.
+
+    Parameters
+    ----------
+    position_array : np.ndarray
+        Array of shape (n_frames, n_individuals, n_space)
+        containing the tracks of the bounding boxes' centroids.
+        It will be converted to a :py:class:`xarray.DataArray` object
+        named "position".
+    shape_array : np.ndarray
+        Array of shape (n_frames, n_individuals, n_space)
+        containing the shape of the bounding boxes. The shape of a bounding
+        box is its width (extent along the x-axis of the image) and height
+        (extent along the y-axis of the image). It will be converted to a
+        :py:class:`xarray.DataArray` object named "shape".
+    confidence_array : np.ndarray, optional
+        Array of shape (n_frames, n_individuals) containing
+        the confidence scores of the bounding boxes. If None (default), the
+        confidence scores are set to an array of NaNs. It will be converted
+        to a :py:class:`xarray.DataArray` object named "confidence".
+    individual_names : list of str, optional
+        List of individual names for the tracked bounding boxes in the video.
+        If None (default), bounding boxes are assigned names based on the size
+        of the `position_array`. The names will be in the format of `id_<N>`,
+        where <N>  is an integer from 0 to `position_array.shape[1]-1` (i.e.,
```

Comment:
reminder that if you want monospace formatting in rst you need double backticks.

Reply:
unless you deliberately wanted italics.